app/models/grda_warehouse/ch_enrollment.rb

```diff
@@ -74,5 +74,323 @@ def self.update_existing!
+      15 => {
+        title: 'Did you stay longer than 90 days?',
+        descriptions: ['If 1 (yes), CONTINUE processing on line 16.', 'If 0 (no), CH = NO. STOP processing.'],
+      },
+      16 => {
+        title: 'On the night before did you stay on the streets, ES or SH',
+        descriptions: ['If 1 (yes), CONTINUE processing on line 17.', 'If 0 (no), CH = NO. STOP processing.'],
+      },
+      17 => {
+        title: 'Days since approximate start date',
+        descriptions: ['If > 365 days, CH = YES. STOP processing.', 'If missing or less than 365 days before [project start date], CONTINUE processing on line 18.'],
+      },
+      18 => {
+        title: 'Number of times homeless',
+        descriptions: ['If four or more times, CONTINUE processing on line 19.', 'If 1, 2, or 3 times, CH = NO. STOP processing.', 'If 8 or 9 then CH = DK/R. STOP processing', 'If 99 then CH = missing. STOP processing'],
+      },
+      19 => {
+        title: 'Total months homeless',
+        descriptions: ['If >= 12, CH = YES. STOP processing.', 'If 1, 2, or 3 times, CH = NO. STOP processing.', 'If 8 or 9 then CH = DK/R. STOP processing', 'If 99 then CH = missing. STOP processing'],
+      },
+      21 => {
+        title: 'Prior Living Situation (3.917B Temporary, Permanent, and other Situations:)',
+        descriptions: ['If 29, 14, 2, 32, 36, 35, 28, 19, 3, 31, 33, 34, 10, 20, 21, 11, 8, 9, 99, CONTINUE processing on line 22.'],
+      },
+      22 => {
+        title: 'Did you stay less than 7 nights?',
+        descriptions: ['If 1 (yes), CONTINUE processing on line 23.', 'If 0 (no), CH = NO. STOP processing.'],
+      },
+      23 => {
+        title: 'On the night before did you stay on the streets, ES or SH',
+        descriptions: ['If 1 (yes), CONTINUE processing on line 24.', 'If 0 (no), CH = NO. STOP processing.'],
+      },
+      24 => {
+        title: 'Days since approximate start date',
+        descriptions: ['If > 365 days, CH = YES. STOP processing.', 'If missing or less than 365 days before [project start date], CONTINUE processing on line 25.'],
+      },
+      25 => {
+        title: 'Number of times homeless',
+        descriptions: ['If four or more times, CONTINUE processing on line 26.', 'If 1, 2, or 3 times, CH = NO. STOP processing.', 'If 8 or 9 then CH = DK/R. STOP processing', 'If 99 then CH = missing. STOP processing'],
+      },
+      26 => {
+        title: 'Total months homeless',
+        descriptions: ['If >= 12, CH = YES. STOP processing.', 'If 1 to 11 months, CH = NO. STOP processing.', 'If 8 or 9 then CH = DK/R. STOP processing', 'If 99 then CH = missing. STOP processing'],
+      },
+
+    }.freeze
+
+    def self.ch_at_entry_matrix(enrollment)
+      chronically_homeless_at_start_steps(enrollment).map do |s|
+        line_number = s[:line]
+        step = CH_AT_DATE_STEP_DESCRIPTIONS[line_number]
+        [line_number, step[:title], s[:display_value], s[:result], step[:descriptions]]
+      end
+    end
```

Comment:
Here we map each executed result to the description information, using the `line` number.